src/reflect/scala/reflect/internal/Types.scala

```diff
@@ -696,21 +698,23 @@ trait Types
      *  }}}
      */
     def memberInfo(sym: Symbol): Type = {
-//      assert(sym ne NoSymbol, this)
+      //      assert(sym ne NoSymbol, this)
       sym.info.asSeenFrom(this, sym.owner)
     }
 
     /** The type of `sym`, seen as a member of this type. */
-    def memberType(sym: Symbol): Type = sym.tpeHK match {
-      case OverloadedType(_, alts) => OverloadedType(this, alts)
+    def memberType(sym: Symbol): Type = sym match {
+      case meth: MethodSymbol =>
+        meth.typeAsMemberOf(this)
+      case _ =>
+        computeMemberType(sym)
+    }
+
+    def computeMemberType(sym: Symbol): Type = sym.tpeHK match {
+      case OverloadedType(_, alts) =>
+        OverloadedType(this, alts)
       case tp =>
-        // Correct caching is nearly impossible because `sym.tpeHK.asSeenFrom(pre, sym.owner)`
```

Comment:
The first thing I'm going to type tomorrow is `// is this comment worth keeping?`  I hope I don't accidentally read it six months from now, because it would blow my brain i.e. mind.  The other one I should plant for my future self is `// what were you thinking?`